File: apps/api/schemas/fotos_credencial.py
from pydantic import BaseModel
from typing import Optional, List
from enum import Enum

class EstadoFoto(str, Enum):
    PENDIENTE = "PENDIENTE"
    APROBADA = "APROBADA"
    RECHAZADA = "RECHAZADA"

class FotoCredencialResponse(BaseModel):
    id: str
    alumnoId: str
    fotoUrl: str
    comentario: Optional[str] = None
    estado: EstadoFoto
    feedback: Optional[str] = None
    evaluadorId: Optional[str] = None
    uploadedAt: str
    updatedAt: Optional[str] = None
    
    # Datos adicionales del alumno (opcional)
    alumnoNombre: Optional[str] = None
    alumnoApellido: Optional[str] = None
    alumnoDni: Optional[str] = None
    empresaNombre: Optional[str] = None

class UploadFotoResponse(BaseModel):
    success: bool
    foto: FotoCredencialResponse
    message: str

class ListFotosResponse(BaseModel):
    fotos: List[FotoCredencialResponse]
    total: int

class EvaluarFotoRequest(BaseModel):
    estado: EstadoFoto
    feedback: Optional[str] = None

File: apps/web/lib/api/fotos-credencial.ts
import { api } from '../api-client';

export type EstadoFoto = 'PENDIENTE' | 'APROBADA' | 'RECHAZADA';

export interface FotoCredencial {
    id: string;
    alumnoId: string;
    fotoUrl: string;
    comentario: string | null;
    estado: EstadoFoto;
    feedback: string | null;
    evaluadorId: string | null;
    uploadedAt: string;
    updatedAt: string | null;
}

export const fotosCredencialApi = {
    async miFoto(alumnoId: string): Promise<FotoCredencial | null> {
        try {
            return await api.get(`/fotos-credencial/alumno/${alumnoId}`);
        } catch (error: any) {
            // El backend devuelve 404 con detail "No se encontró foto para este alumno"
            // cuando el alumno todavía no subió ninguna; api-client no expone el status
            // HTTP, así que distinguimos por el mensaje.
            if (/no se encontr/i.test(error?.message || '')) {
                return null;
            }
            throw error;
        }
    },

    async subirFoto(alumnoId: string, file: File): Promise<{ success: boolean; foto: FotoCredencial; message: string }> {
        const formData = new FormData();
        formData.append('file', file);
        formData.append('alumnoId', alumnoId);
        return api.post('/fotos-credencial/upload', formData);
    },
};
